pkgs/servers/http/nginx/generic.nix

```diff
@@ -60,8 +60,25 @@ stdenv.mkDerivation {
 
   NIX_CFLAGS_COMPILE = [ "-I${libxml2.dev}/include/libxml2" ] ++ optional stdenv.isDarwin "-Wno-error=deprecated-declarations";
 
+  configurePlatforms = [];
+
   preConfigure = (concatMapStringsSep "\n" (mod: mod.preConfigure or "") modules);
 
+  patches = stdenv.lib.optionals (stdenv.hostPlatform != stdenv.buildPlatform) [
+    (fetchpatch {
+      url = "https://raw.githubusercontent.com/openwrt/packages/master/net/nginx/patches/102-sizeof_test_fix.patch";
+      sha256 = "0i2k30ac8d7inj9l6bl0684kjglam2f68z8lf3xggcc2i5wzhh8a";
+    })
+    (fetchpatch {
+      url = "https://raw.githubusercontent.com/openwrt/packages/master/net/nginx/patches/101-feature_test_fix.patch";
+      sha256 = "0v6890a85aqmw60pgj3mm7g8nkaphgq65dj4v9c6h58wdsrc6f0y";
+    })
+    (fetchpatch {
+      url = "https://raw.githubusercontent.com/openwrt/packages/master/net/nginx/patches/103-sys_nerr.patch";
+      sha256 = "0s497x6mkz947aw29wdy073k8dyjq8j99lax1a1mzpikzr4rxlmd";
+    })
+  ];
```

Comment:
@fpletz @thoughtpolice do you find these patches acceptable w.r.t. cross-compiling or do you prefer excluding nginx here?

Reply:
I'm a bit confused by the first patch but the other two look fine to me. But no objection from me though as all of those patches come from OpenWRT and the first one has been in OpenWRT for a while.

Reply:
It searches in the binary for the result sizeof. This allows to get the size of the data structure in the compile test without executing the non-native binary. The text around `sizeof` is just used to create a search pattern.

Reply:
With a bit cleanup they might have a chance to get in upstream.

Reply:
Thanks! Makes sense now.